core/contracts/financial-templates/implementation/demo/DepositBox.sol

```diff
@@ -0,0 +1,346 @@
+
+    // Represents a single caller's deposit box. All collateral is held by this contract.
+    struct DepositBoxData {
+        // Requested amount of collateral, denominated in quote asset of the price identifier.
+        // Example: If the price identifier is wETH-USD, and the `withdrawalRequestAmount = 100`, then
+        // this represents a withdrawal request for 100 USD worth of wETH.
+        FixedPoint.Unsigned withdrawalRequestAmount;
+        // Timestamp of the latest withdrawal request. A withdrawal request is pending if `requestPassTimestamp != 0`.
+        uint256 requestPassTimestamp;
+        // Raw collateral value. This value should never be accessed directly -- always use _getFeeAdjustedCollateral().
+        // To add or remove collateral, use _addCollateral() and _removeCollateral().
+        FixedPoint.Unsigned rawCollateral;
+    }
+
+    // Maps addresses to their deposit boxes. Each address can have only one position.
+    mapping(address => DepositBoxData) public depositBoxes;
+
+    // Unique identifier for DVM price feed ticker.
+    bytes32 public priceIdentifier;
+
+    // Similar to the rawCollateral in DepositBoxData, this value should not be used directly.
+    // _getFeeAdjustedCollateral(), _addCollateral() and _removeCollateral() must be used to access and adjust.
+    FixedPoint.Unsigned public rawTotalDepositBoxCollateral;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event NewDepositBox(address indexed user);
+    event EndedDepositBox(address indexed user);
+    event Deposit(address indexed user, uint256 indexed collateralAmount);
+    event RequestWithdrawal(address indexed user, uint256 indexed collateralAmount, uint256 requestPassTimestamp);
+    event RequestWithdrawalExecuted(
+        address indexed user,
+        uint256 indexed collateralAmount,
+        uint256 exchangeRate,
+        uint256 requestPassTimestamp
+    );
+    event RequestWithdrawalCanceled(
+        address indexed user,
+        uint256 indexed collateralAmount,
+        uint256 requestPassTimestamp
+    );
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier noPendingWithdrawal(address user) {
+        _depositBoxHasNoPendingWithdrawal(user);
+        _;
+    }
+
+    /**
+     * @notice Construct the DepositBox.
+     * @param _collateralAddress ERC20 token to be deposited.
+     * @param _finderAddress UMA protocol Finder used to discover other protocol contracts.
+     * @param _priceIdentifier registered in the DVM, used to price the ERC20 deposited.
+     * The price identifier consists of a "base" asset and a "quote" asset. The "base" asset corresponds to the collateral ERC20
+     * currency deposited into this account, and it is denominated in the "quote" asset on withdrawals.
```

Comment:
this is a great comment on the `_priceIdentifier` param but as this is a tutorial example perhaps add an example `_priceIdentifier`?